src/hierarchy/Animal.ts

```diff
@@ -87,6 +87,12 @@ export default abstract class Animal implements ICanWalk {
     }
     return true
   }
+  public grow(): void {
+    this.age += 1
+    if (this.age >= AGE_TO_BE_ADULT) {
```

Comment:
maybe  age needs a setter that does this check for adult, so we dont need to pass adult or not but setting the check takes care of that on its own